msgbus.go

```diff
@@ -167,30 +180,44 @@ func (mbus *MsgBusWriter_Kafka) Status() int {
 func (mbus *MsgBusWriter_Kafka) MessageWrite(msg string) error {
 	var err error = nil
 
-	if mbus.status != StatusOpen {
+	mbus.handleLock.Lock()
+	defer mbus.handleLock.Unlock()
+
+	if (mbus.status != StatusOpen) {
 		err = fmt.Errorf("ERROR: Attempting MessageWrite() on a closed connection.")
 		return err
 	}
 
-	kmsg := &sarama.ProducerMessage{Topic: mbus.kafkaTopic,
-		Value:     sarama.StringEncoder(msg),
-		Timestamp: time.Now(),
+	if (__testmode) {
+		log.Printf("TestMode: Sending message: %s",msg)
+		return nil
 	}
 
-	if __testmode {
-		fmt.Println("TestMode: Sending message:", kmsg)
-	} else {
-		if mbus.blocking == NonBlocking {
-			(*mbus.kafkaNBProducer).Input() <- kmsg
-			select {
-			case err := <-(*mbus.kafkaNBProducer).Errors():
-				fmt.Println("ERROR on async write channel:", err)
-			default:
-				err = nil
-			}
+	if (mbus.blocking == Blocking) {
+		deliveryChan := make(chan kafka.Event)
+
+		err = mbus.kafkaProducer.Produce(&kafka.Message{
+			TopicPartition: kafka.TopicPartition{Topic: &mbus.kafkaTopic, Partition: kafka.PartitionAny},
+			Value:          []byte(msg),
+		}, deliveryChan)
+
+		e := <-deliveryChan
+		m := e.(*kafka.Message)
+
+		if m.TopicPartition.Error != nil {
+			log.Printf("Kafka message delivery failed: %v\n", m.TopicPartition.Error)
```

Comment:
prepend with "ERROR"

Reply:
also print some meta data like : topic.   We probably dont want to print the message as those could be big and sensitive.   Any other meta data we can expose or should expose?

Reply:
Fixed.